sentry-graphql-java/src/main/java/io/sentry/graphql/SentryDataFetcherExceptionHandler.java

```diff
@@ -0,0 +1,36 @@
+package io.sentry.graphql;
+
+import graphql.execution.DataFetcherExceptionHandler;
+import graphql.execution.DataFetcherExceptionHandlerParameters;
+import graphql.execution.DataFetcherExceptionHandlerResult;
+import io.sentry.HubAdapter;
+import io.sentry.IHub;
+import io.sentry.util.Objects;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Captures exceptions that occur during data fetching, passes them to Sentry and invokes a delegate
+ * exception handler.
+ */
+public final class SentryDataFetcherExceptionHandler implements DataFetcherExceptionHandler {
+  private final @NotNull IHub hub;
+  private final DataFetcherExceptionHandler delegate;
+
+  public SentryDataFetcherExceptionHandler(
+      final @NotNull IHub hub, final @NotNull DataFetcherExceptionHandler delegate) {
+    this.hub = Objects.requireNonNull(hub, "hub is required");
+    this.delegate = Objects.requireNonNull(delegate, "delegate is required");
+  }
+
+  public SentryDataFetcherExceptionHandler(final @NotNull DataFetcherExceptionHandler delegate) {
+    this(HubAdapter.getInstance(), delegate);
+  }
+
+  @Override
+  @SuppressWarnings("deprecation")
+  public DataFetcherExceptionHandlerResult onException(
+      final @NotNull DataFetcherExceptionHandlerParameters handlerParameters) {
+    hub.captureException(handlerParameters.getException());
```

Comment:
should we capture every exception? should we offer maybe an option to turn it off then?
I remember not doing it on SentryOkHttpInterceptor because the user might want to do it themselves based on their own logic.

cc @bruno-garcia

Reply:
Perhaps you mean `BeforeSpanCallback` in OkHttp interceptor? We can add it here too.

Reply:
thats about capturing an exception and not really the span callback, but well remembered, the before span callback could also be a follow up tho

Reply:
Here's where `hints` could be helpful.
As we would capture the exception with the `handlersParameters` as a hint. So if users want to filter some out, they could access that data in the beforeSend or eventProcessor and do they customization/dropping there

Reply:
`hint` added.